src/components/Geogebra/GeogebraMergedAdminView.js

```diff
@@ -0,0 +1,452 @@
+    // the applet. Caption style means that objects are
+    // named by their captions.
+    CAPTION_STYLE: 3,
+
+    POINT: 'point',
+    TEXTFIELD: 'textfield',
+    NUMERIC: 'numeric',
+
+    /**
+     * GGB commands have the following
+     * format: <object-label>:<command-body>.
+     *
+     * @param {String} label
+     * @param {String} cmdStrBody
+     */
+    getCommand(label, cmdStrBody) {
+        return `${label}:${cmdStrBody}`;
+    },
+};
+
+class GeogebraMergedAdminView {
+    constructor(params, workshopIds) {
+        this.params = {
+            showAlgebraInput: true,
+            showToolBarHelp: false,
+            showMenubar: true,
+            enableLabelDrags: false,
+            showResetIcon: true,
+            showToolbar: false,
+            allowStyleBar: false,
+            useBrowserForJS: true,
+            enableShiftDragZoom: true,
+            errorDialogsActive: true,
+            enableRightClick: false,
+            enableCAS: false,
+            enable3d: false,
+            isPreloader: false,
+            screenshotGenerator: false,
+            preventFocus: false,
+            ...params,
+        };
+
+        this.log = this.params.log;
+
+        this.workshopIds = workshopIds;
+        this.initializeReactionsToServerEvents();
+        // eslint-disable-next-line no-undef
+        this.appletContainer = new GGBApplet(this.params);
+    }
+
+    inject(callback) {
+        window.ggbOnInit = () => {
+            this.ggbOnInit();
+            callback();
+        };
+
+        this.appletContainer.inject(this.params.container, 'auto');
+    }
+
+    ggbOnInit() {
+        this.applet = this.appletContainer.getAppletObject();
+        this.isInitialized = true;
+        this.loadWorkshops();
+
+        this.centerView();
+    }
+
+    getXML() {
+        return this.applet.getXML();
+    }
+
+    setXML(xml) {
+        this.applet.setXML(xml);
+    }
+
+    evalXML(xml) {
+        this.applet.evalXML(xml);
+    }
+
+    evalCommand(command) {
+        this.applet.evalCommand(command);
+    }
+
+    getObjectNumber() {
+        return this.applet.getObjectNumber();
+    }
+
+    /**
+     * Binds server events to local callbacks; load workshop meta-data
+     * and elements and initializes applet.
+     */
+    initializeReactionsToServerEvents() {
+        // [onAppletReady] is called by [StudentClient] when applet finished
+        // initializing.
+        this.loadWorkshops = async () => {
+            this.log.debug('loadWorkshops');
+
+            await this.workshopIds.map(
+                async (workshopId, iter) => this.loadWorkshopState(workshopId, iter),
+            );
+
+            this.log.debug('this.workshopIds', this.workshopIds);
+        };
+
+        this.initializeCallbacks = async () => {
+            this.log.debug('Loading initializeCallbacks');
+
+            this.createdListener = (element) => {
+                const pos = this.workshopIds.indexOf(element.workshop);
```

Comment:
See below - we don't need to use `indexOf` (4 occurences in this file).

Reply:
I need the position in array, not only Boolean.

Reply:
I need the position in array, not only Boolean.